src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/WritableShreddedDocument.java

```diff
@@ -304,6 +297,28 @@ public void shredVector(JsonPath path, ArrayNode vector) {
       queryVectorValues = arrayVals;
     }
 
+    @Override
+    public void shredVector(JsonPath path, ObjectNode vector) {
+      // vector data is added only to queryVectorValues and exists keys index
+      addKey(path);
+
+      String binaryString = vector.get(JsonExtensionType.BINARY.encodedName()).asText();
+
+      byte[] binaryPayload;
+      binaryPayload = Base64Variants.MIME_NO_LINEFEEDS.decode(binaryString);
+
+      int numFloats = binaryPayload.length / 4;
+      float[] floatArray = new float[numFloats];
+      ByteBuffer byteBuffer = ByteBuffer.wrap(binaryPayload);
+
+      for (int i = 0; i < numFloats; i++) {
+        int intBits = byteBuffer.getInt(); // Get next 4 bytes as an int
+        floatArray[i] = Float.intBitsToFloat(intBits); // Convert int to float
+      }
```

Comment:
The logic assumes the binary could be decoded correctly; how could we verify if the binary has some problem?

Reply:
As per our discussion, I think 3 main possibilities:

1. Base64 decoding fails with exception (bad character etc)
2. Length decoded is not divisible by 4 
3. Number of floats does not match vector definition.

(1) and (2) can be checked here; (3) would probably need to be validated at a later point.
Should at least test with integration tests to see what happens with (3); Database will verify for sure.